core/metrics-spi/src/main/java/software/amazon/awssdk/metrics/meter/NoOpCounter.java

```diff
@@ -0,0 +1,52 @@
+package software.amazon.awssdk.metrics.meter;
+
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * A NoOp implementation of the {@link Counter} metric
+ */
+@SdkPublicApi
+public enum NoOpCounter implements Counter<Number> {
```

Comment:
Not sure if you are still in the process of updating PR, but this is also using enum for singleton.